File: common/src/main/java/com/codeloom/common/SubmissionEvent.java
package com.codeloom.common;

import java.util.UUID;

public record SubmissionEvent(
        UUID submissionId,
        UUID userId,
        long problemId,
        String code,
        String language,
        Long executionTimeLimitMs,
        Long memoryUsageLimitBytes) {
    public SubmissionEvent(UUID submissionId, UUID userId, long problemId, String code, String language) {
        this(submissionId, userId, problemId, code, language, null, null);
    }
}

File: executor/src/main/kotlin/com/codeloom/executor/event/SubmissionEvent.kt
package com.codeloom.executor.event

import java.util.*

data class SubmissionEvent(
    val submissionId: UUID,
    val userId: UUID,
    val problemId: Long,
    val code: String,
    val language: String,
    val executionTimeLimitMs: Long? = null,
    val memoryUsageLimitBytes: Long? = null,
)
